mvc/models/model-binding.rst

```diff
@@ -1,8 +1,105 @@
-.. include:: /../common/stub-topic.txt
+Model Binding
+=============
 
-|stub-icon| Model Binding Request Data
-======================================
+By `Rachel Appel <http://github.com/rachelappel>`_
 
-.. include:: /../common/stub-notice.txt
+In this article
 
-.. _issue: https://github.com/aspnet/Docs/issues/114
+- `Introduction to model binding`_
+- `How model binding works`_
+- `Customize default model binding behavior with attributes`_
+- `Binding formatted data from the request body`_ 
+- `Resources`_
+
+Introduction to model binding
+-----------------------------
+Model binders are objects in the ASP.NET MVC framework that map data from HTTP requests to action method parameters. The parameters may be simple types such as strings, integers, or floats, or they may be complex objects representing an enterprise-sized model. This is a great feature of MVC because mapping incoming data to a counterpart is an often repeated scenario, regardless of size or complexity of the data. MVC solves this problem by abstracting binding away so developers don't have to keep rewriting a slightly different version of that same code in every app. In previous versions of ASP, including Classic ASP and WebForms, you would extract the HTTP POST or GET values manually, validate the data, and then send that data off to the database. Not much of the process was automated, so you would often see code similar the following example that captures two form values, puts them in variables and TextBox objects, storing them for their journey to the database:
+
+.. literalinclude:: model-binding/sample/src/MVCMovie/retro-sample.txt
+   :language: html
+
+Although the sample shows just two fields and doesn't even include validation, it takes only a moment to realize that the above code will quickly become repetive and cumbersome to write, even when it's just a few fields. Real world apps often have many fields combined with complex structures. Admittedly, this kind of mapping code is a bit boring to write as well. There are details on what's behind ASP.NET MVC's model binding abstraction in the next section, "How model binding works". 
+
+How model binding works
+-----------------------
+When ASP.NET MVC receives an incoming HTTP request, it routes it to a specific action method of a controller. It determines which action method to run based on what is in the route data, then it binds values from the HTTP request to that action method's parameters. For example, consider the following URL:
+
+`http://contoso.com/movies/edit/2`
+
+The default model binder compares the sections of the URL with the pattern defined in the ``RouteValues`` collection. Put a breakpoint in any action method to see the contents of the ``RouteData`` object at run time in a debug window, as illustrated here: 
+
```

Comment:
Great stuff. We should try to include debug examples like this. How can we encourage others to do this?